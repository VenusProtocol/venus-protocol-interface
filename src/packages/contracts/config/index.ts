/* eslint-disable import/no-extraneous-dependencies */
import { abi as IsolatedPoolComptrollerAbi } from '@venusprotocol/isolated-pools/artifacts/contracts/Comptroller.sol/Comptroller.json';
import { abi as JumpRateModelV2Abi } from '@venusprotocol/isolated-pools/artifacts/contracts/JumpRateModelV2.sol/JumpRateModelV2.json';
import { abi as PoolLensAbi } from '@venusprotocol/isolated-pools/artifacts/contracts/Lens/PoolLens.sol/PoolLens.json';
import { abi as PoolRegistryAbi } from '@venusprotocol/isolated-pools/artifacts/contracts/Pool/PoolRegistry.sol/PoolRegistry.json';
import { abi as RewardsDistributorAbi } from '@venusprotocol/isolated-pools/artifacts/contracts/Rewards/RewardsDistributor.sol/RewardsDistributor.json';
import { abi as VBep20Abi } from '@venusprotocol/isolated-pools/artifacts/contracts/VToken.sol/VToken.json';
import isolatedPoolsMainnetDeployments from '@venusprotocol/isolated-pools/deployments/bscmainnet.json';
import isolatedPoolsTestnetDeployments from '@venusprotocol/isolated-pools/deployments/bsctestnet.json';
import { abi as ResilientOracleAbi } from '@venusprotocol/oracle/artifacts/contracts/ResilientOracle.sol/ResilientOracle.json';
import resilientOracleMainnetDeployments from '@venusprotocol/oracle/deployments/bscmainnet/ResilientOracle.json';
import resilientOracleTestnetDeployments from '@venusprotocol/oracle/deployments/bsctestnet/ResilientOracle.json';
import { abi as MainPoolComptrollerAbi } from '@venusprotocol/venus-protocol/artifacts/contracts/Comptroller/Diamond/DiamondConsolidated.sol/DiamondConsolidated.json';
import { abi as GovernorBravoDelegateAbi } from '@venusprotocol/venus-protocol/artifacts/contracts/Governance/GovernorBravoDelegate.sol/GovernorBravoDelegate.json';
import { abi as JumpRateModelAbi } from '@venusprotocol/venus-protocol/artifacts/contracts/InterestRateModels/JumpRateModel.sol/JumpRateModel.json';
import { abi as VenusLensAbi } from '@venusprotocol/venus-protocol/artifacts/contracts/Lens/VenusLens.sol/VenusLens.json';
import { abi as SwapRouterAbi } from '@venusprotocol/venus-protocol/artifacts/contracts/Swap/SwapRouter.sol/SwapRouter.json';
import { abi as Bep20Abi } from '@venusprotocol/venus-protocol/artifacts/contracts/Tokens/BEP20Interface.sol/BEP20Interface.json';
import { abi as VaiAbi } from '@venusprotocol/venus-protocol/artifacts/contracts/Tokens/VAI/VAI.sol/VAI.json';
import { abi as VaiControllerAbi } from '@venusprotocol/venus-protocol/artifacts/contracts/Tokens/VAI/VAIController.sol/VAIController.json';
import { abi as VrtAbi } from '@venusprotocol/venus-protocol/artifacts/contracts/Tokens/VRT/VRT.sol/VRT.json';
import { abi as VrtConverterAbi } from '@venusprotocol/venus-protocol/artifacts/contracts/Tokens/VRT/VRTConverter.sol/VRTConverter.json';
import { abi as XvsAbi } from '@venusprotocol/venus-protocol/artifacts/contracts/Tokens/XVS/XVS.sol/XVS.json';
import { abi as XvsVestingAbi } from '@venusprotocol/venus-protocol/artifacts/contracts/Tokens/XVS/XVSVesting.sol/XVSVesting.json';
import { abi as VaiVaultAbi } from '@venusprotocol/venus-protocol/artifacts/contracts/Vault/VAIVault.sol/VAIVault.json';
import { abi as XvsStoreAbi } from '@venusprotocol/venus-protocol/artifacts/contracts/XVSVault/XVSStore.sol/XVSStore.json';
import { abi as XvsVaultAbi } from '@venusprotocol/venus-protocol/artifacts/contracts/XVSVault/XVSVault.sol/XVSVault.json';
import venusProtocolMainnetDeployments from '@venusprotocol/venus-protocol/networks/mainnet.json';
import venusProtocolTestnetDeployments from '@venusprotocol/venus-protocol/networks/testnet.json';
import { type ContractInterface } from 'ethers';
import { ChainId } from 'types';

import MaximillionAbi from './externalAbis/Maximillion.json';
import multicall3Abi from './externalAbis/Multicall3.json';
import pancakePairV2Abi from './externalAbis/PancakePairV2.json';
import vBnbAbi from './externalAbis/VBnb.json';
import XsequenceMulticall from './externalAbis/XsequenceMulticall.json';

export interface UniqueContractConfig {
  name: string;
  abi: ContractInterface;
  address: Partial<{
    [chainId in ChainId]: string;
  }>;
}

export interface GenericContractConfig {
  name: string;
  abi: ContractInterface;
}

export interface SwapRouterContractConfig {
  name: string;
  abi: ContractInterface;
  address: Partial<{
    [chainId in ChainId]: {
      [comptrollerContractAddress: string]: string;
    };
  }>;
}

export type ContractConfig =
  | UniqueContractConfig
  | GenericContractConfig
  | SwapRouterContractConfig;

export const contracts: ContractConfig[] = [
  // Unique contracts
  {
    name: 'VenusLens',
    abi: VenusLensAbi,
    address: {
      [ChainId.BSC_TESTNET]: venusProtocolTestnetDeployments.Contracts.VenusLens,
      [ChainId.BSC_MAINNET]: venusProtocolMainnetDeployments.Contracts.VenusLens,
    },
  },
  {
    name: 'PoolLens',
    abi: PoolLensAbi,
    address: {
      [ChainId.BSC_TESTNET]: isolatedPoolsTestnetDeployments.contracts.PoolLens.address,
      [ChainId.BSC_MAINNET]: isolatedPoolsMainnetDeployments.contracts.PoolLens.address,
    },
  },
  {
    name: 'PoolRegistry',
    abi: PoolRegistryAbi,
    address: {
      [ChainId.BSC_TESTNET]: isolatedPoolsTestnetDeployments.contracts.PoolRegistry_Proxy.address,
      [ChainId.BSC_MAINNET]: isolatedPoolsMainnetDeployments.contracts.PoolRegistry_Proxy.address,
    },
  },
  {
    name: 'MainPoolComptroller',
    abi: MainPoolComptrollerAbi,
    address: {
      [ChainId.BSC_TESTNET]: venusProtocolTestnetDeployments.Contracts.Unitroller,
      [ChainId.BSC_MAINNET]: venusProtocolMainnetDeployments.Contracts.Unitroller,
    },
  },
  {
    name: 'VaiController',
    abi: VaiControllerAbi,
    address: {
      [ChainId.BSC_TESTNET]: venusProtocolTestnetDeployments.Contracts.VaiUnitroller,
      [ChainId.BSC_MAINNET]: venusProtocolMainnetDeployments.Contracts.VaiUnitroller,
    },
  },
  {
    name: 'VaiVault',
    abi: VaiVaultAbi,
    address: {
      [ChainId.BSC_TESTNET]: venusProtocolTestnetDeployments.Contracts.VAIVaultProxy,
      [ChainId.BSC_MAINNET]: venusProtocolMainnetDeployments.Contracts.VAIVaultProxy,
    },
  },
  {
    name: 'XvsVault',
    abi: XvsVaultAbi,
    address: {
      [ChainId.BSC_TESTNET]: venusProtocolTestnetDeployments.Contracts.XVSVaultProxy,
      [ChainId.BSC_MAINNET]: venusProtocolMainnetDeployments.Contracts.XVSVaultProxy,
    },
  },
  {
    name: 'XvsStore',
    abi: XvsStoreAbi,
    address: {
      [ChainId.BSC_TESTNET]: venusProtocolTestnetDeployments.Contracts.XVSStore,
      [ChainId.BSC_MAINNET]: venusProtocolMainnetDeployments.Contracts.XVSStore,
    },
  },
  {
    name: 'GovernorBravoDelegate',
    abi: GovernorBravoDelegateAbi,
    address: {
      [ChainId.BSC_TESTNET]: venusProtocolTestnetDeployments.Contracts.GovernorBravoDelegator,
      [ChainId.BSC_MAINNET]: venusProtocolMainnetDeployments.Contracts.GovernorBravoDelegator,
    },
  },
  {
    name: 'XvsVesting',
    abi: XvsVestingAbi,
    address: {
      [ChainId.BSC_TESTNET]: venusProtocolTestnetDeployments.Contracts.XVSVestingProxy,
      [ChainId.BSC_MAINNET]: venusProtocolMainnetDeployments.Contracts.XVSVestingProxy,
    },
  },
  {
    name: 'VrtConverter',
    abi: VrtConverterAbi,
    address: {
      [ChainId.BSC_TESTNET]: venusProtocolTestnetDeployments.Contracts.VRTConverterProxy,
      [ChainId.BSC_MAINNET]: venusProtocolMainnetDeployments.Contracts.VRTConverterProxy,
    },
  },
  {
    name: 'Maximillion',
    abi: MaximillionAbi,
    address: {
      [ChainId.BSC_TESTNET]: '0xF3a34e06015e019D6154a0f1089f695B27122f50',
      [ChainId.BSC_MAINNET]: '0x5efA1e46F4Fd738FF721F5AebC895b970F13E8A1',
    },
  },
  {
    name: 'XsequenceMulticall',
    abi: XsequenceMulticall,
    address: {
      [ChainId.BSC_MAINNET]: '0xd130B43062D875a4B7aF3f8fc036Bc6e9D3E1B3E',
      [ChainId.BSC_TESTNET]: '0xd130B43062D875a4B7aF3f8fc036Bc6e9D3E1B3E',
    },
  },
  {
    name: 'Multicall3',
    abi: multicall3Abi,
    address: {
      [ChainId.BSC_TESTNET]: '0xca11bde05977b3631167028862be2a173976ca11',
      [ChainId.BSC_MAINNET]: '0xca11bde05977b3631167028862be2a173976ca11',
    },
  },
  {
    name: 'ResilientOracle',
    abi: ResilientOracleAbi,
    address: {
      [ChainId.BSC_TESTNET]: resilientOracleTestnetDeployments.address,
      [ChainId.BSC_MAINNET]: resilientOracleMainnetDeployments.address,
    },
  },
  // Generic Contracts
  {
    name: 'IsolatedPoolComptroller',
    abi: IsolatedPoolComptrollerAbi,
  },
  {
    name: 'JumpRateModel',
    abi: JumpRateModelAbi,
  },
  {
    name: 'JumpRateModelV2',
    abi: JumpRateModelV2Abi,
  },
  {
    name: 'RewardsDistributor',
    abi: RewardsDistributorAbi,
  },
  {
    name: 'VBep20',
    abi: VBep20Abi,
  },
  {
    name: 'VBnb',
    abi: vBnbAbi,
  },
  {
    name: 'Bep20',
    abi: Bep20Abi,
  },
  {
    name: 'Xvs',
    abi: XvsAbi,
  },
  {
    name: 'Vai',
    abi: VaiAbi,
  },
  {
    name: 'Vrt',
    abi: VrtAbi,
  },
  {
    name: 'PancakePairV2',
    abi: pancakePairV2Abi,
  },
  // SwapRouter contract
  {
    name: 'SwapRouter',
    abi: SwapRouterAbi,
    address: {
      [ChainId.BSC_TESTNET]: {
        // Main pool
        [venusProtocolTestnetDeployments.Unitroller.address.toLowerCase()]:
          venusProtocolTestnetDeployments.Contracts.SwapRouterCorePool,
        // Isolated pools
        [isolatedPoolsTestnetDeployments.contracts.Comptroller_StableCoins.address.toLowerCase()]:
          isolatedPoolsTestnetDeployments.contracts.SwapRouter_StableCoins.address,
        [isolatedPoolsTestnetDeployments.contracts.Comptroller_Tron.address.toLowerCase()]:
          isolatedPoolsTestnetDeployments.contracts.SwapRouter_Tron.address,
        [isolatedPoolsTestnetDeployments.contracts.Comptroller_GameFi.address.toLowerCase()]:
          isolatedPoolsTestnetDeployments.contracts.SwapRouter_GameFi.address,
        [isolatedPoolsTestnetDeployments.contracts.Comptroller_DeFi.address.toLowerCase()]:
          isolatedPoolsTestnetDeployments.contracts.SwapRouter_DeFi.address,
        [isolatedPoolsTestnetDeployments.contracts.Comptroller_LiquidStakedBNB.address.toLowerCase()]:
          isolatedPoolsTestnetDeployments.contracts.SwapRouter_LiquidStakedBNB.address,
      },
      [ChainId.BSC_MAINNET]: {
        // Main pool
        [venusProtocolMainnetDeployments.Unitroller.address.toLowerCase()]:
          venusProtocolMainnetDeployments.Contracts.SwapRouterCorePool,
        // Isolated Pools
        [isolatedPoolsMainnetDeployments.contracts.Comptroller_Stablecoins.address.toLowerCase()]:
          isolatedPoolsMainnetDeployments.contracts.SwapRouter_Stablecoins.address,
        [isolatedPoolsMainnetDeployments.contracts.Comptroller_Tron.address.toLowerCase()]:
          isolatedPoolsMainnetDeployments.contracts.SwapRouter_Tron.address,
        [isolatedPoolsMainnetDeployments.contracts.Comptroller_GameFi.address.toLowerCase()]:
          isolatedPoolsMainnetDeployments.contracts.SwapRouter_GameFi.address,
        [isolatedPoolsMainnetDeployments.contracts.Comptroller_DeFi.address.toLowerCase()]:
          isolatedPoolsMainnetDeployments.contracts.SwapRouter_DeFi.address,
        [isolatedPoolsMainnetDeployments.contracts.Comptroller_LiquidStakedBNB.address.toLowerCase()]:
          isolatedPoolsMainnetDeployments.contracts.SwapRouter_LiquidStakedBNB.address,
      },
    },
  },
];
