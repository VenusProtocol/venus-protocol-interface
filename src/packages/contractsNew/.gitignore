# Automatically generated files
/getters
/infos
