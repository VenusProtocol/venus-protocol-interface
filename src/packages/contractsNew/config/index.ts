import { abi as isolatedPoolComptrollerAbi } from '@venusprotocol/isolated-pools/artifacts/contracts/Comptroller.sol/Comptroller.json';
import isolatedPoolsMainnetDeployments from '@venusprotocol/isolated-pools/deployments/bscmainnet.json';
import isolatedPoolsTestnetDeployments from '@venusprotocol/isolated-pools/deployments/bsctestnet.json';
import { abi as mainPoolComptrollerAbi } from '@venusprotocol/venus-protocol/artifacts/contracts/Comptroller/Comptroller.sol/Comptroller.json';
import { abi as venusLensAbi } from '@venusprotocol/venus-protocol/artifacts/contracts/Lens/VenusLens.sol/VenusLens.json';
import { abi as swapRouterAbi } from '@venusprotocol/venus-protocol/artifacts/contracts/Swap/SwapRouter.sol/SwapRouter.json';
import venusProtocolMainnetDeployments from '@venusprotocol/venus-protocol/networks/mainnet.json';
import venusProtocolTestnetDeployments from '@venusprotocol/venus-protocol/networks/testnet.json';
import { type ContractInterface } from 'ethers';
import { ChainId } from 'types';

export interface UniqueContractConfig {
  name: string;
  abi: ContractInterface;
  address: Partial<{
    [chainId in ChainId]: string;
  }>;
}

export interface GenericContractConfig {
  name: string;
  abi: ContractInterface;
}

export interface SwapRouterContractConfig {
  name: string;
  abi: ContractInterface;
  address: Partial<{
    [chainId in ChainId]: {
      [comptrollerContractAddress: string]: string;
    };
  }>;
}

export type ContractConfig =
  | UniqueContractConfig
  | GenericContractConfig
  | SwapRouterContractConfig;

export const contracts: ContractConfig[] = [
  // Unique contracts
  {
    name: 'VenusLens',
    abi: venusLensAbi,
    address: {
      [ChainId.BSC_TESTNET]: venusProtocolTestnetDeployments.Contracts.VenusLens,
      [ChainId.BSC_MAINNET]: venusProtocolMainnetDeployments.Contracts.VenusLens,
    },
  },
  {
    name: 'MainPoolComptroller',
    abi: mainPoolComptrollerAbi,
    address: {
      [ChainId.BSC_TESTNET]: venusProtocolTestnetDeployments.Contracts.Unitroller,
      [ChainId.BSC_MAINNET]: venusProtocolMainnetDeployments.Contracts.Unitroller,
    },
  },
  // TODO: add all contracts
  // Generic Contracts
  {
    name: 'IsolatedPoolComptroller',
    abi: isolatedPoolComptrollerAbi,
  },
  // TODO: add all contracts
  // SwapRouter contract
  {
    name: 'SwapRouter',
    abi: swapRouterAbi,
    address: {
      [ChainId.BSC_TESTNET]: {
        // Main pool
        [venusProtocolTestnetDeployments.Unitroller.address.toLowerCase()]:
          venusProtocolTestnetDeployments.Contracts.SwapRouterCorePool,
        // Isolated pools
        [isolatedPoolsTestnetDeployments.contracts.Comptroller_StableCoins.address.toLowerCase()]:
          isolatedPoolsTestnetDeployments.contracts.SwapRouter_StableCoins.address,
        [isolatedPoolsTestnetDeployments.contracts.Comptroller_Tron.address.toLowerCase()]:
          isolatedPoolsTestnetDeployments.contracts.SwapRouter_Tron.address,
        [isolatedPoolsTestnetDeployments.contracts.Comptroller_GameFi.address.toLowerCase()]:
          isolatedPoolsTestnetDeployments.contracts.SwapRouter_GameFi.address,
        [isolatedPoolsTestnetDeployments.contracts.Comptroller_DeFi.address.toLowerCase()]:
          isolatedPoolsTestnetDeployments.contracts.SwapRouter_DeFi.address,
        [isolatedPoolsTestnetDeployments.contracts.Comptroller_LiquidStakedBNB.address.toLowerCase()]:
          isolatedPoolsTestnetDeployments.contracts.SwapRouter_LiquidStakedBNB.address,
      },
      [ChainId.BSC_MAINNET]: {
        // Main pool
        [venusProtocolMainnetDeployments.Unitroller.address.toLowerCase()]:
          venusProtocolMainnetDeployments.Contracts.SwapRouterCorePool,
        // Isolated Pools
        [isolatedPoolsMainnetDeployments.contracts.Comptroller_Stablecoins.address.toLowerCase()]:
          isolatedPoolsMainnetDeployments.contracts.SwapRouter_Stablecoins.address,
        [isolatedPoolsMainnetDeployments.contracts.Comptroller_Tron.address.toLowerCase()]:
          isolatedPoolsMainnetDeployments.contracts.SwapRouter_Tron.address,
        [isolatedPoolsMainnetDeployments.contracts.Comptroller_GameFi.address.toLowerCase()]:
          isolatedPoolsMainnetDeployments.contracts.SwapRouter_GameFi.address,
        [isolatedPoolsMainnetDeployments.contracts.Comptroller_DeFi.address.toLowerCase()]:
          isolatedPoolsMainnetDeployments.contracts.SwapRouter_DeFi.address,
        [isolatedPoolsMainnetDeployments.contracts.Comptroller_LiquidStakedBNB.address.toLowerCase()]:
          isolatedPoolsMainnetDeployments.contracts.SwapRouter_LiquidStakedBNB.address,
      },
    },
  },
];

export default contracts;
