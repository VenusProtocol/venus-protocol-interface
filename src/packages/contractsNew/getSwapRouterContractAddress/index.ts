import addresses from 'packages/contractsNew/infos/addresses';
import { ChainId } from 'types';

export type GetSwapRouterContractAddressInput = {
  comptrollerContractAddress: string;
  chainId: ChainId;
};

const getSwapRouterContractAddress = ({
  comptrollerContractAddress,
  chainId,
}: GetSwapRouterContractAddressInput) => {
  const swapRouterContractAddresses = addresses.SwapRouter[chainId];
  const sanitizedComptrollerAddress =
    comptrollerContractAddress.toLowerCase() as keyof typeof swapRouterContractAddresses;

  return swapRouterContractAddresses?.[sanitizedComptrollerAddress] as string | undefined;
};

export default getSwapRouterContractAddress;
