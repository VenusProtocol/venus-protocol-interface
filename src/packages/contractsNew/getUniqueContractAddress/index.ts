import addresses from 'packages/contractsNew/infos/addresses';
import { UniqueContractName } from 'packages/contractsNew/infos/types';
import { ChainId } from 'types';

export type GetUniqueContractAddressInput = {
  name: UniqueContractName;
  chainId: ChainId;
};

const getUniqueContractAddress = ({ name, chainId }: GetUniqueContractAddressInput) =>
  addresses[name][chainId];

export default getUniqueContractAddress;
