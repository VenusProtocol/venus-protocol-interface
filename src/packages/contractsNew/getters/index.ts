/* Automatically generated file, do not update manually */
export * from './venusLens';
export * from './poolLens';
export * from './poolRegistry';
export * from './mainPoolComptroller';
export * from './vaiController';
export * from './vaiVault';
export * from './xvsVault';
export * from './xvsStore';
export * from './governorBravoDelegate';
export * from './xvsVesting';
export * from './vrtConverter';
export * from './maximillion';
export * from './xsequenceMulticall';
export * from './multicall3';
export * from './resilientOracle';
export * from './isolatedPoolComptroller';
export * from './jumpRateModel';
export * from './jumpRateModelV2';
export * from './rewardsDistributor';
export * from './vToken';
export * from './vBnb';
export * from './bep20';
export * from './xvs';
export * from './vai';
export * from './vrt';
export * from './pancakePairV2';
export * from './swapRouter';
