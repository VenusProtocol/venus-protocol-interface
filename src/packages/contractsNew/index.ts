// Export contract getter functions and hooks
export * from './getters';
// Export types
export * from './infos/contractTypes';
export * from './infos/types';
// Export addresses
export { default as addresses } from './infos/addresses';
