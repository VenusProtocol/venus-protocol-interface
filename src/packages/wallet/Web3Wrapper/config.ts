import { configureChains, createConfig } from 'wagmi';
import { CoinbaseWalletConnector } from 'wagmi/connectors/coinbaseWallet';
import { InjectedConnector } from 'wagmi/connectors/injected';
import { MetaMaskConnector } from 'wagmi/connectors/metaMask';
import { WalletConnectConnector } from 'wagmi/connectors/walletConnect';
import { jsonRpcProvider } from 'wagmi/providers/jsonRpc';

import localConfig from 'config';
import { WALLET_CONNECT_PROJECT_ID } from 'constants/walletConnect';
import { chains } from 'packages/wallet/chains';
import { ChainId } from 'types';

import { BinanceWalletConnector } from './binanceWalletConnector';

const { publicClient, webSocketPublicClient } = configureChains(
  chains,
  [
    jsonRpcProvider({
      rpc: chain => localConfig.rpcUrls[chain.id as ChainId],
    }),
  ],
  {
    stallTimeout: 3000, // Time after which a request is dimmed stalled and another provider is used
    batch: {
      multicall: false, // Disable wagmi's multicall feature as we wrap the provider with a 0xsequence multicall provider instead
    },
  },
);

const config = createConfig({
  autoConnect: false,
  publicClient,
  webSocketPublicClient,
  connectors: [
    new InjectedConnector({ chains }),
    new MetaMaskConnector({ chains }),
    new WalletConnectConnector({
      chains,
      options: {
        projectId: WALLET_CONNECT_PROJECT_ID,
        showQrModal: true,
      },
    }),
    new CoinbaseWalletConnector({
      chains,
      options: {
        appName: 'Venus',
      },
    }),
    new BinanceWalletConnector({
      chains,
    }),
  ],
});

export default config;
