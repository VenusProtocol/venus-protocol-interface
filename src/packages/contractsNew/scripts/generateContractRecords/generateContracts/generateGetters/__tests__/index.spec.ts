import { ContractConfig } from 'packages/contractsNew/config';
import Vi from 'vitest';

import writeFile from 'utilities/writeFile';

import generateGetters from '..';

vi.mock('utilities/writeFile');

const fakeContractConfigs: ContractConfig[] = [
  {
    name: 'PoolLens',
    abi: [],
    address: {},
  },
  {
    name: 'IsolatedPoolComptroller',
    abi: [],
  },
  {
    name: 'SwapRouter',
    abi: [],
    address: {},
  },
];

describe('generateGetters', () => {
  it('calls writeFile with the right arguments', () => {
    generateGetters({
      outputDirectoryPath: 'fake/output/director/path',
      contractConfigs: fakeContractConfigs,
    });

    expect(writeFile).toHaveBeenCalledTimes(fakeContractConfigs.length);

    fakeContractConfigs.forEach((_fakeContractConfig, index) =>
      expect((writeFile as Vi.Mock).mock.calls[index]).toMatchSnapshot(),
    );
  });
});
