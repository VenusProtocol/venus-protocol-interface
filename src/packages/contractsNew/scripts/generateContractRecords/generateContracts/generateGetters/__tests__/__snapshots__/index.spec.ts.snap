// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`generateGetters > calls writeFile with the right arguments 1`] = `
[
  {
    "content": "import abi from 'packages/contractsNew/generated/infos/abis/PoolLens.json';
import { PoolLens } from 'packages/contractsNew/generated/infos/contractTypes';

import { uniqueContractAddressGetterGenerator } from 'packages/contractsNew/utilities/uniqueContractAddressGetterGenerator';
import { uniqueContractAddressGetterHookGenerator } from 'packages/contractsNew/utilities/uniqueContractAddressGetterHookGenerator';
import { uniqueContractGetterGenerator } from 'packages/contractsNew/utilities/uniqueContractGetterGenerator';
import { uniqueContractGetterHookGenerator } from 'packages/contractsNew/utilities/uniqueContractGetterHookGenerator';

export const getPoolLensContractAddress = uniqueContractAddressGetterGenerator({
  name: 'PoolLens',
});

export const useGetPoolLensContractAddress = uniqueContractAddressGetterHookGenerator({
  getter: getPoolLensContractAddress,
});

export const getPoolLensContract = uniqueContractGetterGenerator<PoolLens>({
  name: 'PoolLens',
  abi,
});

export const useGetPoolLensContract = uniqueContractGetterHookGenerator({
  getter: getPoolLensContract,
});
",
    "outputPath": "fake/output/director/path/poolLens.ts",
  },
]
`;

exports[`generateGetters > calls writeFile with the right arguments 2`] = `
[
  {
    "content": "import abi from 'packages/contractsNew/generated/infos/abis/IsolatedPoolComptroller.json';
import { IsolatedPoolComptroller } from 'packages/contractsNew/generated/infos/contractTypes';

import { genericContractGetterGenerator } from 'packages/contractsNew/utilities/genericContractGetterGenerator';
import { genericContractGetterHookGenerator } from 'packages/contractsNew/utilities/genericContractGetterHookGenerator';

export const getIsolatedPoolComptrollerContract = genericContractGetterGenerator<IsolatedPoolComptroller>({
  abi,
});

export const useGetIsolatedPoolComptrollerContract = genericContractGetterHookGenerator({
  getter: getIsolatedPoolComptrollerContract,
});
",
    "outputPath": "fake/output/director/path/isolatedPoolComptroller.ts",
  },
]
`;

exports[`generateGetters > calls writeFile with the right arguments 3`] = `
[
  {
    "content": "import abi from 'packages/contractsNew/generated/infos/abis/SwapRouter.json';
import { SwapRouter } from 'packages/contractsNew/generated/infos/contractTypes';

import { swapRouterContractGetterGenerator } from 'packages/contractsNew/utilities/swapRouterContractGetterGenerator';
import { swapRouterContractGetterHookGenerator } from 'packages/contractsNew/utilities/swapRouterContractGetterHookGenerator';

export * from 'packages/contractsNew/utilities/useGetSwapRouterContractAddress';
export * from 'packages/contractsNew/utilities/getSwapRouterContractAddress';

export const getSwapRouterContract = swapRouterContractGetterGenerator<SwapRouter>({
  abi,
});

export const useGetSwapRouterContract = swapRouterContractGetterHookGenerator({
  getter: getSwapRouterContract,
});
",
    "outputPath": "fake/output/director/path/swapRouter.ts",
  },
]
`;

exports[`generateGetters > calls writeFile with the right arguments 4`] = `
[
  {
    "content": "export * from './poolLens';
export * from './isolatedPoolComptroller';
export * from './swapRouter';
",
    "outputPath": "fake/output/director/path/index.ts",
  },
]
`;
